apps/opencs/model/world/columnimp.cpp

```diff
@@ -1,28 +1,261 @@
 #include "columnimp.hpp"
 
-CSMWorld::BodyPartRaceColumn::BodyPartRaceColumn(const MeshTypeColumn<ESM::BodyPart> *meshType)
-    : mMeshType(meshType)
-{}
-
-QVariant CSMWorld::BodyPartRaceColumn::get(const Record<ESM::BodyPart> &record) const
+namespace CSMWorld
 {
-    if (mMeshType != NULL && mMeshType->get(record) == ESM::BodyPart::MT_Skin)
+    /* LandMapLodColumn */
+    LandMapLodColumn::LandMapLodColumn()
+        : Column<Land>(Columns::ColumnId_LandMapLodIndex, ColumnBase::Display_String, 0)
     {
-        return QString::fromUtf8(record.get().mRace.c_str());
     }
-    return QVariant(QVariant::UserType);
-}
 
-void CSMWorld::BodyPartRaceColumn::set(Record<ESM::BodyPart> &record, const QVariant &data)
-{
-    ESM::BodyPart record2 = record.get();
+    QVariant LandMapLodColumn::get(const Record<Land>& record) const
+    {
+        const int Size = Land::LAND_GLOBAL_MAP_LOD_SIZE;
+        const Land& land = record.get();
 
-    record2.mRace = data.toString().toUtf8().constData();
+        if (land.isDataLoaded(Land::DATA_WNAM))
+        {
+            // Note: original data is signed
+            const char* rawData = reinterpret_cast<const char*>(&land.mWnam[0]);
+            return QByteArray(rawData, Size);
+        }
+        else
+        {
+            // Return a blank array
+            return QByteArray(Size, 0);
+        }
+    }
 
-    record.setModified(record2);
-}
+    void LandMapLodColumn::set(Record<Land>& record, const QVariant& data)
+    {
+        QByteArray array = data.toByteArray();
+        const signed char* rawData = reinterpret_cast<const signed char*>(array.data());
+        assert (array.count() == Land::LAND_GLOBAL_MAP_LOD_SIZE);
```

Comment:
That looks problematic. We shouldn't shut down the editor in such a case, because it would result in data loss. An exception would be a better option.